src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -95,36 +91,20 @@ bool GetRequest::Wait(int64_t timeout_ms) {
   RAY_CHECK(timeout_ms >= 0 || timeout_ms == -1);
   if (timeout_ms == -1) {
     // Wait forever until all objects are ready.
-    Wait();
+    std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
It would be nice if we could have thread annotation for std::mutex, I could write one.

Reply:
agreed would be nice, or we can standardize on always using absl, it has its own condition variable too

Reply:
absl mute is not good enough from my experience, since it doesnt work with a lot of mutexes

Reply:
For example, tracy is used to report lock acquisition/release timing, which requires std::condition_variable_any